### ISSUE TITLE
Protractor Not Registering Clicks on Elements with `ng-click` with Webpacks

### ISSUE THREAD

**Issue by joshmgrant** [angular/protractor#4475]
Bug report
---

- Node Version: `8.1.2`
- Protractor Version: `5.1.2`
- Angular Version: `1.5.11`
- Browser(s): `Chrome`
- Operating System and Version `MacOS Sierra`

Test file:
```javascript
describe('should click the element', function() {
    it('using ng-click', function(){
                time = typeof time !== 'undefined' ? time : 5000;
                el = element(by.css('[ng-click="vm.addInventory()"]')
                browser.wait(EC.elementToBeClickable(el), time);
                el().click();
    });
});
```

```html
<div class="jm-spec jm-squire text-right jm-clickable" ng-click="vm.addInventory()" ng-switch-when="false" role="button" tabindex="0"><div class="jm-value">ADD<span class="fa fa-plus"></span></div></div>
```
Expected Behaviour: Element is clicked

Actual Behaviour: Element is not clicked, but does not fail on `EC.elementToBeClickable` condition. 

Additional Notes: This appears to occur with other elements with an `ng-click` and this issue only appeared after introducing Webpack into our Angular project. Templates are rendered using Pug.

**Comment by wswebcreation** [comment]
Hmm

Kinda strange. Do you Have a simple example project to reproduceren the bug? 
And do you see the same behavior with other browsers like Firefox. And which version of Chrome and Chrome driver are you using (sorry for all the questions 😉)

**Comment by joshmgrant** [comment]
Unfortunately I can't provide a sample project to reproduce atm. But I did provide Chrome/Chromedriver versions.

**Comment by wswebcreation** [comment]
Sorry, didn't saw that.

And what about other browsers? Without an example it would be hard to debug for is unless more people with an example project have the same issue

**Comment by wswebcreation** [comment]
What I was also thinking, Chrome was updated to 61 20hours ago, could that also be the problem? So what happens s if you downgrade to 60? Just to eliminate that

**Comment by joshmgrant** [comment]
I don't think this is the case since these issues have been in our codebase for ~1 week now. The only change that occurred in that time was introduce Webpack.

**Comment by pradeep636** [comment]
I have seen similar issue. My protractor code perfectly works on chrome but when i change browser to Firefox, it fails. The error i see is that protractor or may be gecko-driver is not waiting for the page load to happen after click() and the next step gets failed.

**Scenario:** click on sign in button and enter email
sign in button was clicked but after that find element failed for email

**Failed: No element found using locator: By(css selector, *[id="email"])**

I used below versions
[00:05:58] I/status - selenium standalone versions available: 3.5.1, 3.5.2, 3.5.3 [last]
[00:05:58] I/status - chromedriver versions available: 2.31, 2.32 [last]
[00:05:58] I/status - geckodriver version available: v0.18.0 [last]
Firefox: 56

Is this a know issue or any settings i need to follow?

**Comment by wswebcreation** [comment]
@pradeep636 

Strange that it works on Chrome and not on Firefox. Are you sure the click is really executed on Firefox, even if you for example add a `browser.sleep(5000)`. In some cases and on some browsers (mostly Firefox and Safari) a click event is only triggered if the exact element is clicked that holds the `ng-click`, have you also checked that?

**Comment by pradeep636** [comment]
@wswebcreation 
I check the button element and could see that it doesn't have ng-click. So might be protractor is not waiting for angular. But if that's the case then the script should fail in chrome too. Is it the chrome driver and gecko driver that are having different logic in waiting after button click?

**Comment by FreshQa** [comment]
@joshmgrant I too am seeing a similar instance, however my OS is Windows 10 Pro. It would appear the click event is being called, but the actual event is off-module. In my case, since the click exposes a SAP flag before progressing to the next step, the scenario fails with a timeout. Same result for Firefox, and we do not support IE/Edge.
